.docker/ci/Dockerfile

```diff
@@ -1,48 +1,89 @@
-# moveit/moveit:melodic-ci
+# moveit/moveit:crystal-ci
 # Sets up a base image to use for running Continuous Integration on Travis
+FROM ros:crystal-ros-base-bionic
 
-FROM ros:melodic-ros-base
-MAINTAINER Dave Coleman dave@picknik.ai
+LABEL maintainer="mike@picknik.ai"
 
 ENV TERM xterm
 
 # Setup catkin workspace
-ENV CATKIN_WS=/root/ws_moveit
-WORKDIR $CATKIN_WS
+ENV ROS_WS=/root/ws_moveit
+ENV ROS_DISTRO crystal
+
+WORKDIR $ROS_WS
 
 # Commands are combined in single RUN statement with "apt/lists" folder removal to reduce image size
 # https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers
-RUN \
-    mkdir src && \
-    cd src && \
+
+RUN mkdir src && \
+    cd $ROS_WS/src && \
     #
     # Download moveit source, so that we can get necessary dependencies
-    wstool init --shallow . https://raw.githubusercontent.com/ros-planning/moveit/${ROS_DISTRO}-devel/moveit.rosinstall && \
+    wstool init --shallow . https://raw.githubusercontent.com/ros-planning/moveit2/moveit2-ci/moveit.rosinstall && \
     #
     # Update apt package list as previous containers clear the cache
-    apt-get -qq update && \
-    apt-get -qq dist-upgrade && \
+    apt-get -qq update  && \
+    apt-get -qq dist-upgrade -y && \
     #
     # Install some base dependencies
     apt-get -qq install -y \
         # Some source builds require a package.xml be downloaded via wget from an external location
         wget \
         # Required for rosdep command
         sudo \
-        # Preferred build tools
-        python-catkin-tools \
         clang clang-format-3.9 clang-tidy clang-tools \
-        ccache && \
+        ccache \
+        curl gnupg2 lsb-release \
+        # Some python dependencies for working with ROS2
+        python3-colcon-common-extensions \
+        python3-pip \
+        python-rosdep \
+        python3-vcstool \
+        python3-wstool \
+        python3-rospkg-modules \
+        python3-rospkg-modules \
+        python3-rosdistro-modules \
+        && \
+    #
+    # Upgrade Pip
+    python3 -m pip install --upgrade pip && \
+    #
+    # install some pip packages needed for testing
+    python3 -m pip install -U \
+        argcomplete \
+        flake8 \
+        flake8-blind-except \
+        flake8-builtins \
+        flake8-class-newline \
+        flake8-comprehensions \
+        flake8-deprecated \
+        flake8-docstrings \
+        flake8-import-order \
+        flake8-quotes \
+        # git+https://github.com/lark-parser/lark.git@0.7d \
+        pytest-repeat \
+        pytest-rerunfailures \
+        pytest \
+        pytest-cov \
+        pytest-runner \
+        setuptools \
+        && \
+    #
+    # install Fast-RTPS dependencies
```

Comment:
Shouldn't this be part of the ROS base image?

Reply:
It should be but it seems that it is not the case. Upon inspection `libasio-dev` is already installed in the base image but `libtinyxml2-dev` isn't. I left both of them since they are both included in the source installation instructions.